src/extension/src/types/generationPayloadType.ts

```diff
@@ -1,34 +1,55 @@
 import { API } from "../azure/azure-cosmosDB/cosmosDbModule";
 
-export interface IGenerationPayloadType {
+export interface IGenerationData {
   backendFramework: string;
   frontendFramework: string;
   backendFrameworkLinuxVersion: string;
   pages: any;
   path: string;
   projectName: string;
   projectType: string;
-  services: IServicesGenerationPayload;
+  services: Array<IService>;
 }
 
-export interface IServicesGenerationPayload {
-  appService: IAppServiceGenerationPayload | null;
-  cosmosDB: ICosmosDBGenerationPayload | null;
-}
+export type IService = IAppService|ICosmosDB|IDefaultService;
 
-export interface IAppServiceGenerationPayload {
-  subscription: string;
-  resourceGroup: string;
-  location: string;
-  siteName: string;
+export type IBaseService = {
   internalName: string;
+  serviceName: string;
+  category: SERVICE_CATEGORY;
+  type: SERVICE_TYPE;
+};
+
+//This interface is for services without azure generation (code generation only by cli)
+export interface IDefaultService extends IBaseService {
+  category: SERVICE_CATEGORY.DEFAULT;
+  type: SERVICE_TYPE.DEFAULT;
 }
 
-export interface ICosmosDBGenerationPayload {
+export interface IAzureService extends IBaseService{
+  category: SERVICE_CATEGORY.AZURE;
   subscription: string;
   resourceGroup: string;
   location: string;
-  accountName: string;
+}
+
+export interface IAppService extends IAzureService {
```

Comment:
Not sure but could it make sense to move those interfaces and enums to somewhere else? Or maybe it´s just the name of the file generationPayloadType. The only type that seems tied to "generation" it´s just generationData, right? The rest sounds more generic just related to Services... Maybe have a new file _**serviceType**_ ?